crates/router/src/connector/bluesnap.rs

```diff
@@ -462,16 +571,130 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
         data: &types::PaymentsAuthorizeRouterData,
         res: Response,
     ) -> CustomResult<types::PaymentsAuthorizeRouterData, errors::ConnectorError> {
+        match (data.is_three_ds(), res.headers) {
+            (true, Some(headers)) => {
+                let location = connector_utils::get_http_header("Location", &headers)?;
+                let payment_fields_token = url::Url::parse(location)
+                    .into_report()
+                    .change_context(errors::ConnectorError::URLParsingFailed {
```

Comment:
Instead of parsing this to URL we can parse the string itself. And there is no need for new error type "URLParsingFailed"